instrumentation/camel-2.20/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/AwsConnector.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apachecamel.aws;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.BucketNotificationConfiguration;
+import com.amazonaws.services.s3.model.ObjectListing;
+import com.amazonaws.services.s3.model.QueueConfiguration;
+import com.amazonaws.services.s3.model.S3Event;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.amazonaws.services.s3.model.SetBucketNotificationConfigurationRequest;
+import com.amazonaws.services.sns.AmazonSNSAsyncClient;
+import com.amazonaws.services.sns.model.CreateTopicResult;
+import com.amazonaws.services.sqs.AmazonSQSAsyncClient;
+import com.amazonaws.services.sqs.model.GetQueueAttributesRequest;
+import com.amazonaws.services.sqs.model.PurgeQueueRequest;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.amazonaws.services.sqs.model.SendMessageRequest;
+import io.opentelemetry.instrumentation.test.utils.PortUtils;
+import java.util.Collections;
+import java.util.EnumSet;
+import org.elasticmq.rest.sqs.SQSRestServer;
+import org.elasticmq.rest.sqs.SQSRestServerBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class AwsConnector {
+  private static final Logger logger = LoggerFactory.getLogger(AwsConnector.class);
+  private AmazonSQSAsyncClient sqsClient;
+  private AmazonS3Client s3Client;
+  private AmazonSNSAsyncClient snsClient;
+  private SQSRestServer sqsRestServer;
+
+  static AwsConnector elasticMq() {
+    AwsConnector awsConnector = new AwsConnector();
+    int sqsPort = PortUtils.findOpenPort();
+    awsConnector.sqsRestServer =
```

Comment:
Nit: instead of setting these fields like that, you could make them `final` and set the via a constructor.